### ISSUE TITLE
"Did not receive valid JSON" when json: true is passed in

### ISSUE THREAD

**Issue by halkeye** [PlasmaPower/assert-request#2]
```
$ cat test_app.js
/* eslint-env node,mocha */
var express = require('express');
var app = express();

app.post('/', function (req, res) {
  res.json({
    'response_type': 'ephemeral',
    'text': 'hi there\nhi\n'
  });
});

var AssertRequest = require('assert-request');
var request = AssertRequest(app.listen());

describe('basic', function () {
  it('json.post', function () {
    return request.post('/', {
      body: { 'key': 'value' },
      json: true
    })
      .type('application/json')
      .status(200)
      .json({
        'response_type': 'ephemeral',
        'text': 'hi there\nhi\n'
      });
  });
});
```

`json: true` in request will automatically encode the sending and receiving body.

One possible fix is to check `if (typeof res.body === 'object')` in mixins/json.js but I'm not sure its the best solution so havn't submitted a PR.

**Comment by PlasmaPower** [comment]
Fixed via 67d62f0 (forgot to put that in the commit message)

**Comment by halkeye** [comment]
woo, time to update my tests
